### ISSUE TITLE
Investigate RPC v0.5 starknet-rs errors

### ISSUE THREAD

**Issue by Mirko-von-Leipzig** [eqlabs/pathfinder#1495]
Some starknet-rs tests fail for RPC v0.5; specifically the `starknet_traceTransaction` and `starknet_getTransactionReceipt` methods fail for every transaction variant. They all fail with a typically unhelpful serde decoding error: 
```
data did not match any variant of untagged enum JsonRpcResponse
```
This indicates we (or less likely, starknet-rs) have a bad encoding for our responses here. We already have #1494 for a missing field for L1 handler receipt. We should compare our spec to the output again. Definitely for receipts we re-used the v0.4 implementation so we would have missed all changes here.

List of test failures
```
jsonrpc_get_transaction_receipt_declare
jsonrpc_get_transaction_receipt_deploy
jsonrpc_get_transaction_receipt_deploy_account
jsonrpc_get_transaction_receipt_invoke
jsonrpc_get_transaction_receipt_invoke_reverted
jsonrpc_get_transaction_receipt_l1_handler
jsonrpc_trace_declare
jsonrpc_trace_deploy_account
jsonrpc_trace_invoke
jsonrpc_trace_invoke_reverted
jsonrpc_trace_l1_handler
```

**Comment by kkovaacs** [comment]
We have several issues in pathfinder 0.9.4:

- transaction traces should include a `type` property which we don't have
- pathfinder uses the `DELEGATE` call type which the specification does not allow (but as discussed this is a specification error, added now in the 0.5.1 spec)
- builtin property names are wrong in the EXECUTION_RESOURCES type (missing `_applications`)
- L1 handler transaction receipt is missing the `message_hash` property.

**Comment by Mirko-von-Leipzig** [comment]
Fixed in v0.9.5
